github/actions_artifacts_test.go

```diff
@@ -0,0 +1,243 @@
+	})
+
+	artifact, resp, err := client.Actions.GetArtifact(context.Background(), "o", "r", 1)
+	if err == nil {
+		t.Errorf("Expected HTTP 404 response")
+	}
+	if got, want := resp.Response.StatusCode, http.StatusNotFound; got != want {
+		t.Errorf("Actions.GetArtifact return status %d, want %d", got, want)
+	}
+	if artifact != nil {
+		t.Errorf("Actions.GetArtifact return %+v, want nil", artifact)
+	}
+}
+
+func TestActionsSerivice_DownloadArtifact(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/artifacts/1/zip", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		w.Header()["Location"] = []string{"https://location.com/artifact"}
+		w.WriteHeader(http.StatusNoContent)
+	})
+
+	_, err := client.Actions.DownloadArtifact(context.Background(), "o", "r", 1, "zip")
+	if err != nil {
+		t.Errorf("Actions.DownloadArtifact returned error: %v", err)
+	}
+}
+
+func TestActionsService_DownloadArtifact_invalidOwner(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, err := client.Actions.DownloadArtifact(context.Background(), "%", "r", 1, "zip")
+	testURLParseError(t, err)
+}
+
+func TestActionsService_DownloadArtifact_invalidRepo(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, err := client.Actions.DownloadArtifact(context.Background(), "%", "r", 1, "zip")
+	testURLParseError(t, err)
+}
+
+func TestActionsService_DownloadArtifact_notFound(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/artifacts/1/zip", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		w.WriteHeader(http.StatusNotFound)
+	})
+
+	resp, err := client.Actions.DownloadArtifact(context.Background(), "o", "r", 1, "zip")
+	if err == nil {
+		t.Errorf("Expected HTTP 404 response")
+	}
+	if got, want := resp.Response.StatusCode, http.StatusNotFound; got != want {
+		t.Errorf("Actions.DownloadArtifact return status %d, want %d", got, want)
+	}
+}
+
+func TestActionsService_DownloadArtifact_invalidArchiveFormat(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, err := client.Actions.DownloadArtifact(context.Background(), "o", "r", 1, "tgz")
+	if err == nil {
+		t.Errorf("Expected HTTP 404 response")
+	}
+}
+
+func TestActionsService_DeleteArtifact(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/artifacts/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "DELETE")
+	})
+
+	_, err := client.Actions.DeleteArtifact(context.Background(), "o", "r", 1)
+	if err != nil {
+		t.Errorf("Actions.DeleteArtifact return error: %v", err)
+	}
+}
+
+func TestActionsService_DeleteArtifact_invalidOwner(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, err := client.Actions.DeleteArtifact(context.Background(), "%", "r", 1)
```

Comment:
What is this actually testing?  I think it can be deleted.

Reply:
https://github.com/google/go-github/pull/1414#discussion_r376749484